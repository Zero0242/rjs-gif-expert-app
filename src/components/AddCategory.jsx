import { useState } from "react";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  /**
   * @param {React.ChangeEvent<HTMLInputElement>} event
   */
  const onChange = ({ target }) => {
    setInputValue(target.value);
  };

  /**
   * @param {React.FormEvent<HTMLFormElement>} event
   */
  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    setCategories( inputValue.trim() );
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type='text'
        placeholder='Buscar gifs'
        value={inputValue}
        onChange={onChange}
      />
      ;
    </form>
  );
};
