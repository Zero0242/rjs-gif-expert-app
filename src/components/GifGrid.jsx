import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  const loadgifs = async () => {
    const newImages = await getGifs(category);
    setImages(newImages);
  };

  useEffect(() => {
    loadgifs();
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className='card-grid'>
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
