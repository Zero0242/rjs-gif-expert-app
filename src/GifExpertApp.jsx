import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch", "Dragon Ball"]);

  /**
   * @param {string} value
   * @returns
   */
  const onAddCategory = (value) => {
    if (categories.includes(value)) return;
    setCategories([value, ...categories]);
  };

  return (
    <>
      <h1>Gif Expert App</h1>
      <AddCategory setCategories={onAddCategory} />
      {categories.map((category, index) => (
        <GifGrid 
            category={category} 
            key={category + index} 
        />
      ))}
    </>
  );
};
