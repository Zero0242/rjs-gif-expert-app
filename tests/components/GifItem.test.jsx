import { GifItem } from "../../src/components";
import { render, screen } from "@testing-library/react";

describe("Pruebas en <GifItem />", () => {
  test("Deberia hacer match con el snapshot", () => {
    const url = "https://tenor.com/hIiAum6gFkk.gif";
    const title = "Pruebas";
    const { container } = render(<GifItem title={title} url={url} />);

    screen.debug();
    expect(container).toMatchSnapshot();
  });
});
