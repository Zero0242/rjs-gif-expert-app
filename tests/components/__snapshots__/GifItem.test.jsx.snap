// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Deberia hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pruebas"
      src="https://tenor.com/hIiAum6gFkk.gif"
    />
    <p>
      Pruebas
    </p>
  </div>
</div>
`;
